kv/union_store.go

```diff
@@ -121,7 +129,12 @@ func (us *unionStore) Get(k Key) ([]byte, error) {
 	v, err := us.MemBuffer.Get(k)
 	if IsErrNotFound(err) {
 		if _, ok := us.opts.Get(PresumeKeyNotExists); ok {
-			err = us.markLazyConditionPair(k, nil)
+			e, ok := us.opts.Get(PresumeKeyNotExistsError)
+			if ok {
+				err = us.markLazyConditionPair(k, nil, e.(error))
```

Comment:
Should we check the type conversion here in order to make sure it will not panic?